cmd/curio/rpc/rpc.go

```diff
@@ -244,7 +245,7 @@ func (p *CurioAPI) LogSetLevel(ctx context.Context, subsystem, level string) err
 	return logging.SetLogLevel(subsystem, level)
 }
 
-func ListenAndServe(ctx context.Context, dependencies *deps.Deps, shutdownChan chan struct{}) error {
+func ListenAndServe(ctx context.Context, dependencies *deps.Deps, activeTasks []harmonytask.TaskInterface, shutdownChan chan struct{}) error {
```

Comment:
Can we avoid doing all this if we just add these details to the machine details table?

Reply:
No, this is about knowing which tasks we can get the SP_ID details for and which are "whole system" tasks. This is not a database matter but an introspection of which enabled tasks support this interface.

Reply:
Makes sense.